### PR TITLE
allow to use custom runtime

### LINKED ISSUE #56: How to use with a custom runtime (that also extends SymfonyRuntime)
I have a custom runtime that is like this 

```
<?php                                                                                                     
                                                                                                                                
declare(strict_types=1);                                                                                  
                                                                                                                                         
namespace App\Runtime;                                                                                                                        
                                                                                                                                                       
use App\Config\KmsDotEnvLoader;                                                                           
use Symfony\Component\Dotenv\Dotenv;                                                                                                                        
use Symfony\Component\Runtime\SymfonyRuntime;                                                              
                                                                                                                 
class SymfonyWithKMSRuntime extends SymfonyRuntime                                                                                                                                                      
{
    /**
     * {@inheritdoc}
     *
     * @param array<mixed> $options                                                                                                                        
     */
    public function __construct(array $options = [])
    {
        parent::__construct($options);
        $kmsDotEnvLoader = new KmsDotEnvLoader(new Dotenv());
        $kmsDotEnvLoader->inject(\dirname(__DIR__).'/../'.KmsDotEnvLoader::ENCRYPTED_FILE);
    }
}
```

i.e it reads additional env variables from a kms-encrypted file

however if i try to use the version 0.2.0 wihtout php-fpm, it fails because the BrefRuntime is hardcoded.

Is there a way to do what I want with the bref/symfony-bridge as it is ?  (i.e to get the env variables set-up, I don't mind not using 'my' runtime, as long as the job is done) 

I got it working by moving this piece of code in the constructor of my kernel,  but it degrades a lot the performance ( I think because now it needs to read the configuration at every request and not only cold ones ? )

### PR DESCRIPTION
**Pull request by allan-simon**
close #56

### CODE REVIEW

**Comment by allan-simon** [comment]
now it should be good @mnapoli

**Comment by DaMitchell** [comment]
Is there anything stopping this from being added?

**Comment by mnapoli** [comment]
@DaMitchell what would be your use case?

**Comment by mnapoli** [comment]
Got it. Is it something that could be implemented in `Kernel::boot()` or a similar method?

Is there any reason to implement it in a custom runtime? (I'm trying to be exhaustive and figure out if this is really something worth the maintenance effort)

**Comment by allan-simon** [comment]
in my case it was because the `env` and `debug` attribute used to construct the kernel were overwritten by the dotenv value , and so `Kernel::boot` is already too late

**Comment by t-richard** [comment]
@allan-simon maybe it didn't worded in boot because of something similar to what I found in this PR https://github.com/brefphp/symfony-bridge/pull/37

That said, adding support for any runtime is out of the scope of this bridge for now. 

I fear that if we support that, we end up with people wanting to use runtimes for roadrunner, reactphp or swoole in Lambda for example and that may not work as expected in a Lambda environment.

That being said, if that's a real blocker, we could allow to specify a runtime but only if it's an instance of SymfonyRuntime. 

That way we support use cases like yours but constrain it to something we know works and that we can maintain.

**Comment by allan-simon** [comment]
@t-richard  I've checked, and I think here it was not the issue, but if I remember my investigation (i'ts been a while so I may recall incorrectly) it was really a "by design" , as the Kernel constructor already expect to receive env variables , which mean they need to be ready by the time you instantiate it.

I understand that opening to any kind of custom runtime may widen too much the scope and make it unbearable to support in term of human time,  so your proposition of limiting it to sub-class of SymfonyRuntime sounds like a good compromise to me.

**Review by Nyholm — APPROVED** [review]
I appreciate this.

**Review by mnapoli — CHANGES_REQUESTED** [review]
Thanks, to piggy-back on the comments above let's merge this with a check that the runtime extends `SymfonyRuntime`

**Comment by allan-simon** [comment]
sure no problem

File: src/HandlerResolver.php [hunk 1]
```diff
@@ -114,7 +113,8 @@ private function symfonyContainer(?string $bootstrapFile = null): ContainerInter
             if ($projectDir) {
                 $options['project_dir'] = $projectDir;
             }
-            $runtime = new BrefRuntime($options);
+            $runtimeClass = $_SERVER['APP_RUNTIME'] ?? $_ENV['APP_RUNTIME'] ?? '\\Bref\\SymfonyBridge\\Runtime\\BrefRuntime';
+            $runtime = new $runtimeClass($options);
```
Comment (allan-simon) [inline src/HandlerResolver.php:117]:
```suggestion
            $runtime = new $runtimeClass($options);
            if (!$runtime instanceof \Symfony\Component\Runtime\SymfonyRuntime) {
                throw new \RuntimeException("The runtime class '$runtimeClass' must extend Symfony\Component\Runtime\SymfonyRuntime.");
            }
}
```

**Comment by allan-simon** [comment]
@mnapoli  done, I've tested locally by monkeypatching my vendor

**Review by mnapoli — APPROVED** [review]

**Comment by Nyholm** [comment]
I've rebased the PR and fixed the CI issues

**Review by Nyholm — APPROVED** [review]

**Pull request merged**